File: 4. Median of Two Sorted Arrays.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """  
        m, n = len(nums1), len(nums2)
        nums = nums1 + nums2
        nums.sort()
        return (nums[(m + n)/2]+nums[(m + n)/2 - 1])/2.0 if not (n + m) % 2 else nums[(m + n - 1)/2]

File: 4. Median of Two Sorted Arrays.c
int compInc(const void *a, const void *b) {  
    return *(int *)a - *(int *)b;  
}  

double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int* nums = (int *) malloc(sizeof(int) * (nums1Size + nums2Size)), i;
    for (i = 0; i < nums1Size; i++)
        nums[i] = nums1[i];  // save nums1 to nums
    for (; i < nums1Size + nums2Size; i++)
        nums[i] = nums2[i - nums1Size];  // save nums2 to nums
    qsort(nums, nums1Size + nums2Size, sizeof(nums[0]), compInc);  // sort nums
    return (nums1Size + nums2Size) % 2 ? (double)(nums[(nums1Size + nums2Size - 1)/2]) : (double)(nums[(nums1Size + nums2Size)/2] + nums[(nums1Size + nums2Size)/2-1])/2.0;
}
